src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs

```diff
@@ -247,6 +240,995 @@ public bool NeedsRefresh()
+                                    FeatureManagementConstants.To,
+                                    settingKey,
+                                    reader.TokenType.ToString(),
+                                    JsonTokenType.Number.ToString());
+                            }
+
+                            break;
+                        }
+
+                    default:
+                        reader.Skip();
+
+                        break;
+                }
+            }
+
+            return featurePercentileAllocation;
+        }
+
+        private FeatureVariant ParseFeatureVariant(ref Utf8JsonReader reader, string settingKey)
+        {
+            var featureVariant = new FeatureVariant();
+
+            while (reader.Read() && reader.TokenType != JsonTokenType.EndObject)
+            {
+                if (reader.TokenType != JsonTokenType.PropertyName)
+                {
+                    continue;
+                }
+
+                string variantPropertyName = reader.GetString();
+
+                switch (variantPropertyName)
+                {
+                    case FeatureManagementConstants.Name:
+                        {
+                            if (reader.Read() && reader.TokenType == JsonTokenType.String)
+                            {
+                                featureVariant.Name = reader.GetString();
+                            }
+                            else if (reader.TokenType != JsonTokenType.Null)
+                            {
+                                throw CreateFeatureFlagFormatException(
+                                    FeatureManagementConstants.Name,
+                                    settingKey,
+                                    reader.TokenType.ToString(),
+                                    JsonTokenType.String.ToString());
+                            }
+
+                            break;
+                        }
+
+                    case FeatureManagementConstants.ConfigurationReference:
+                        {
+                            if (reader.Read() && reader.TokenType == JsonTokenType.String)
+                            {
+                                featureVariant.ConfigurationReference = reader.GetString();
+                            }
+                            else if (reader.TokenType != JsonTokenType.Null)
+                            {
+                                throw CreateFeatureFlagFormatException(
+                                    FeatureManagementConstants.ConfigurationReference,
+                                    settingKey,
+                                    reader.TokenType.ToString(),
+                                    JsonTokenType.String.ToString());
+                            }
+
+                            break;
+                        }
+
+                    case FeatureManagementConstants.ConfigurationValue:
+                        {
+                            if (reader.Read())
```

Comment:
Do we need to check if reader.TokenType == JsonTokenType.StartObject ?

Reply:
And throw exception if its not an object or null

Reply:
I don't think we should need that check, `configuration_value` can be any valid json so a number, boolean, string, etc. would all be valid.